### PR TITLE
Feature/init ruby script

### PR DESCRIPTION
**Pull request by zenangst**
I made a small ruby script that helps you with setting up a new pod.

It does the following;

- Prompts you for your pod name
- Does a search and replace for <PODNAME> & <PODDESCRIPTION>
- Renames the .podspec
- Removes `.git`
- Runs `git init` and make an initial commit.
- Adds a `git remote`

So now you just need to clone the thing and run `ruby init.rb`

### CODE REVIEW

**Comment by zenangst** [comment]
You could take it a step further and have it rename the folder that you clone to match the pod name.

**Comment by NSElvis** [comment]
Or even hook it up to GitHub API to create the repo! (2.0?)

**Comment by zenangst** [comment]
That would be top notch!

File: init.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+require 'fileutils'
+
+folder_path = Dir.pwd
+
+print 'pod name > '
+pod_name = gets.chop
+print 'pod description > '
```
Comment (NSElvis) [inline init.rb]: Can we remove the description to make it even smaller? You usually don't know the entire details of what you're making until you make it.

Comment (zenangst) [inline init.rb]: Sure, but I tend to include descriptions in mine.

**Comment by NSElvis** [comment]
This. Is. Amazeballs.

Fantastic work @zenangst :)

File: init.rb (hunk 1 above)
Comment (zenangst) [inline init.rb]: But that make sense.

Comment (NSElvis) [inline init.rb]: @zenangst Wait! I meant summary is fine, but drop description.

**Comment by zenangst** [comment]
@NSElvis should be good now I think

**Comment by timkurvers** [comment]
Would probably be good to add a shebang and chmod the file, to allow executing: `./init.rb`

File: init.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+file_names = Dir["#{Dir.pwd}/**/*.*"]
+ignored_file_types = ['.xccheckout',
+                      '.xcodeproj',
+                      '.xcworkspace',
+                      '.xcuserdatad',
+                      '.xcuserstate',
+                      '.rb']
+
+file_names.each do |file_name|
+    if !ignored_file_types.include?(File.extname(file_name))
```
Comment (timkurvers) [inline init.rb]: Should strictly speaking be two space indentation here.

Comment (zenangst) [inline init.rb]: I'll fix :+1:

**Comment by zenangst** [comment]
@timkurvers great idea, I'll fix!

**Comment by zenangst** [comment]
Are we good? :)

File: init.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/ruby
```
Comment (timkurvers) [inline init.rb]: Do we wanna force the system ruby here? Or rather `#!/usr/bin/env ruby` which might use an rbenv/rvm installed version.

**Comment by zenangst** [comment]
@timkurvers Fixed, thanks for your feedback ;)

File: init.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env ruby
+
+require 'fileutils'
+
+folder_path = Dir.pwd
+
+print 'pod name > '
+pod_name = gets.chop
+
+file_names = Dir["#{Dir.pwd}/**/*.*"]
```
Comment (timkurvers) [inline init.rb]: Teehee, missed one :angel:

**Comment by zenangst** [comment]
@NSElvis, @timkurvers I think that should do it.

**Comment by zenangst** [comment]
@NSElvis, @timkurvers I think that should do it.

**Comment by timkurvers** [comment]
:+1:

**Comment by timkurvers** [comment]
Might need to document this in the README, and then.. uhh.. remove that bit afterwards?

Never mind, that might be complicated! :smile:

**Comment by NSElvis** [comment]
@zenangst I'll try your magic!

**Comment by NSElvis** [comment]
@zenangst https://github.com/hyperoslo/pod-template/pull/8

**Comment by NSElvis** [comment]
@zenangst can't find the full cake, so here you got

:cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake: :cake:

**Comment by jgorset** [comment]
I like you.

**Pull request merged**